src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -2570,7 +2570,27 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Property
 
         Public Overrides Function CreateErrorTypeSymbol(container As INamespaceOrTypeSymbol, name As String, arity As Integer) As INamedTypeSymbol
-            Return New ExtendedErrorTypeSymbol(DirectCast(container, NamespaceOrTypeSymbol), name, arity)
+            If name Is Nothing Then
+                Throw New ArgumentNullException(NameOf(name))
+            End If
+
+            If arity < 0 Then
+                Throw New ArgumentException(NameOf(arity))
+            End If
+
+            Return New ExtendedErrorTypeSymbol(container.EnsureVbSymbolOrNothing(Of NamespaceOrTypeSymbol)(NameOf(container)), name, arity)
+        End Function
+
+        Public Overrides Function CreateErrorNamespaceSymbol(container As INamespaceSymbol, name As String) As INamespaceSymbol
+            If container Is Nothing Then
```

Comment:
consider pulling hte error checking up to the base type (to prevent code duplication), and only have the language specific code in the derived types.